cms/articles/models.py

```diff
@@ -414,11 +452,60 @@ def related_data(self, request: "HttpRequest") -> "TemplateResponse":
         )
         return response
 
+    def as_featured_article_macro_data(self, request: "HttpRequest") -> dict[str, Any]:
+        """Returns data formatted for the onsFeaturedArticle Nunjucks/Jinja2 macro."""
+        data = {
+            "title": {
+                "url": self.get_url(request),
+                "text": self.listing_title or self.display_title,
+            },
+            "metadata": {
+                "text": self.label,
+            },
+            "description": self.main_points_summary,
+        }
+
+        if self.release_date:
+            data["metadata"]["date"] = {
+                "prefix": _("Release date"),
+                "showPrefix": True,
+                "iso": self.release_date.isoformat(),
+                "short": ons_date_format(datetime.combine(self.release_date, datetime.min.time()), "DATE_FORMAT"),
+            }
+
+        if self.featured_chart:
+            chart_block = self.featured_chart[0]  # pylint: disable=unsubscriptable-object
+            block_instance = chart_block.block
+            block_value = chart_block.value
+
+            if isinstance(block_instance, BaseVisualisationBlock):
+                data["chart"] = block_instance.get_component_config(block_value)
+            else:
+                raise ValueError(f"Unsupported block type: {type(block_instance)}")
```

Comment:
question: should this not move to the Page.clean or the admin form clean? That is, we want to surface these issues before publication

Reply:
I'm not sure that an editor would really be able to add an unsupported block type. This is here more as a placeholder because locally I have these lines in a git stash:

```diff
+            elif isinstance(block_instance, ONSEmbedBlock):
+                ...  # TODO Return component data for iframe, see https://github.com/ONSdigital/design-system/pull/3641
```

Perhaps the error lines can be removed.

Reply:
If the intention is to prevent us (developers) from shooting our foot, I'd recommend a system check.

Otherwise it may be cleaner to have a sensible fallback as we do not want the topic page to explode for unexplained reasons

Reply:
I have removed these lines in 852f3c9.